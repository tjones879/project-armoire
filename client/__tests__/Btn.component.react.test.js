import React from 'react';
import {shallow, mount, render} from 'enzyme';
import {Btn} from '../src/components/Btn.component.react';

let props, wrapper, node;

let mockEvent = jest.fn();
mockEvent.mockImplementation(() => {
});

beforeEach(() => {
    props = {
        text:'button',
        type:'button',
        id:'BtnID',
        event: mockEvent,
        disabled: true
    }
    wrapper = shallow(<Btn id={props.id} text={props.text} type={props.type} event={props.event} disabled={props.disabled}/>);
    node = wrapper.find(`button#${props.id}`);
});

describe('shallow rendering of Btn', () => {
    test('Props should be sent in properly', () => {
        expect(wrapper.instance().props).toMatchObject(props);
    });
    test('Btn text should match the prop sent in', () => {
        expect(node.text()).toEqual(props.text);
    });
    test('mock button click should happen', () => {
        node.simulate('click');
        expect(mockEvent).toHaveBeenCalled();
    });
});

describe('snapshot testing of Btn component', () => {
    test('snapshot should remain the same', () => {
        expect(wrapper).toMatchSnapshot();
    });
});