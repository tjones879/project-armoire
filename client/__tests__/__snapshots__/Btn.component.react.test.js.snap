// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing of Btn component snapshot should remain the same 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Btn
    disabled={true}
    event={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    id="BtnID"
    style={
      Object {
        "display": "inline-block",
      }
    }
    text="button"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "button",
      "className": undefined,
      "disabled": true,
      "id": "BtnID",
      "onClick": [MockFunction] {
        "calls": Array [
          Array [],
        ],
      },
      "style": Object {
        "display": "inline-block",
      },
      "type": "button",
    },
    "ref": null,
    "rendered": "button",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "button",
        "className": undefined,
        "disabled": true,
        "id": "BtnID",
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
        "style": Object {
          "display": "inline-block",
        },
        "type": "button",
      },
      "ref": null,
      "rendered": "button",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
